### ISSUE TITLE
Running phpunit tests from PhpStorm tooling breaks if mutagen is enabled

### ISSUE THREAD

**Issue by cosmicdreams** [ddev/ddev#4697]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Run a Diagnostic and Paste Link Here

<details><description>ddev debug test</description>

```
chrisweber@MacBook-Pro ddev-nomutagen % ddev debug test
Running bash [-c /var/folders/cl/7ss_0jzj2q14mxk7qys2qj980000gp/T/test_ddev.sh] 
======= Existing project config =========
These config files were loaded for project ddev-nomutagen: [/Users/chrisweber/Sites/ddev-nomutagen/.ddev/config.yaml /Users/chrisweber/Sites/ddev-nomutagen/.ddev/config.selenium-standalone-chrome.yaml]
name: ddev-nomutagen
type: drupal10
docroot: web
php_version: 8.1
webserver_type: nginx-fpm
webimage: drud/ddev-webserver:v1.21.4
router_http_port: 80
router_https_port: 443
database: {mariadb 10.4}
mailhog_port: 8025
mailhog_https_port: 8026
phpmyadmin_port: 8036
phpmyadmin_https_port: 8037
project_tld: ddev.site
use_dns_when_possible: true
composer_version: 2
nodejs_version: 16
default_container_timeout: 120
======= Creating dummy project named  tryddevproject-16641 in ../tryddevproject-16641 =========
OS Information: Darwin MacBook-Pro 22.3.0 Darwin Kernel Version 22.3.0: Mon Jan 30 20:38:37 PST 2023; root:xnu-8792.81.3~2/RELEASE_ARM64_T6000 arm64
ProductName:		macOS
ProductVersion:		13.2.1
BuildVersion:		22D68
User information: uid=502(chrisweber) gid=20(staff) groups=20(staff),12(everyone),61(localaccounts),79(_appserverusr),80(admin),81(_appserveradm),98(_lpadmin),702(com.apple.sharepoint.group.2),33(_appstore),100(_lpoperator),204(_developer),250(_analyticsusers),395(com.apple.access_ftp),398(com.apple.access_screensharing),399(com.apple.access_ssh),400(com.apple.access_remote_ae),701(com.apple.sharepoint.group.1)
DDEV version:  ITEM             VALUE                                   
 DDEV version     v1.21.4                                 
 architecture     arm64                                   
 db               drud/ddev-dbserver-mariadb-10.4:v1.21.4 
 dba              phpmyadmin:5                            
 ddev-ssh-agent   drud/ddev-ssh-agent:v1.21.4             
 docker           20.10.23                                
 docker-compose   v2.14.0                                 
 docker-platform  docker-desktop                          
 mutagen          0.16.0                                  
 os               darwin                                  
 router           drud/ddev-router:v1.21.4                
 web              drud/ddev-webserver:v1.21.4             
PROXY settings: HTTP_PROXY='' HTTPS_PROXY='' http_proxy='' NO_PROXY=''
======= DDEV global info =========
Global configuration:
instrumentation-opt-in=true
omit-containers=[]
mutagen-enabled=false
nfs-mount-enabled=false
router-bind-all-interfaces=false
internet-detection-timeout-ms=3000
disable-http2=false
use-letsencrypt=false
letsencrypt-email=
table-style=default
simple-formatting=false
auto-restart-containers=false
use-hardened-images=false
fail-on-hook-fail=false
required-docker-compose-version=
use-docker-compose-from-path=false
project-tld=
xdebug-ide-location=
no-bind-mounts=false
use-traefik=false

======= DOCKER info =========
docker location: lrwxr-xr-x  1 root  wheel  54 Dec  8 13:09 /usr/local/bin/docker -> /Applications/Docker.app/Contents/Resources/bin/docker
Docker Desktop Version: Docker Desktop for Mac 4.17.0 build 99724
docker version: 
Client:
 Cloud integration: v1.0.31
 Version:           20.10.23
 API version:       1.41
 Go version:        go1.18.10
 Git commit:        7155243
 Built:             Thu Jan 19 17:35:19 2023
 OS/Arch:           darwin/arm64
 Context:           default
 Experimental:      true

Server: Docker Desktop 4.17.0 (99724)
 Engine:
  Version:          20.10.23
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.18.10
  Git commit:       6051f14
  Built:            Thu Jan 19 17:31:28 2023
  OS/Arch:          linux/arm64
  Experimental:     false
 containerd:
  Version:          1.6.18
  GitCommit:        2456e983eb9e37e47538f59ea18f2043c9a73640
 runc:
  Version:          1.1.4
  GitCommit:        v1.1.4-0-g5fd4c4d
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
DOCKER_DEFAULT_PLATFORM=notset
======= Mutagen Info =========
Mutagen is installed in ddev, version=0.16.0
--------------------------------------------------------------------------------
Name: core-dev
Identifier: sync_pNkRwebP3T5tVjBivnFWUdTBZAyNxQ5Id9VsSLV33wL
Labels:
	com.ddev.volume-signature: var-run-1677778329
Configuration:
	Synchronization mode: Two Way Resolved
	Maximum allowed entry count: Default (2⁶⁴−1)
	Maximum staging file size: Default (18 EB)
	Symbolic link mode: POSIX Raw
	Ignore VCS mode: Default (Propagate)
	Ignores:
		/.git
		/.tarballs
		/.ddev/db_snapshots
		/.ddev/.importdb*
		.DS_Store
		.idea
		/web/sites/default/files
	Permissions mode: Default (Portable)
Alpha:
	URL: /Users/chrisweber/Sites/core-dev
	Configuration:
		Watch mode: Default (Portable)
		Watch polling interval: Default (10 seconds)
		Probe mode: Default (Probe)
		Scan mode: Default (Accelerated)
		Stage mode: Neighboring
		File mode: Default (0600)
		Directory mode: Default (0700)
		Default file/directory owner: Default
		Default file/directory group: Default
	Connected: No
Beta:
	URL: docker://ddev-core-dev-web/var/www/html
		DOCKER_HOST=unix:///var/run/docker.sock
	Configuration:
		Watch mode: Default (Portable)
		Watch polling interval: Default (10 seconds)
		Probe mode: Default (Probe)
		Scan mode: Default (Accelerated)
		Stage mode: Neighboring
		File mode: Default (0600)
		Directory mode: Default (0700)
		Default file/directory owner: Default
		Default file/directory group: Default
	Connected: No
Status: [Paused]
--------------------------------------------------------------------------------
Name: ddev-mutagen
Identifier: sync_M4cHZaIiqNFrwRkYc0IVw4YcdNyexYdYdUcMwNPnL6A
Labels:
	com.ddev.volume-signature: Users-chrisweber-colima-default-1677786106
Configuration:
	Synchronization mode: Two Way Resolved
	Maximum allowed entry count: Default (2⁶⁴−1)
	Maximum staging file size: Default (18 EB)
	Symbolic link mode: POSIX Raw
	Ignore VCS mode: Default (Propagate)
	Ignores:
		/.git
		/.tarballs
		/.ddev/db_snapshots
		/.ddev/.importdb*
		.DS_Store
		.idea
		/web/sites/default/files
	Permissions mode: Default (Portable)
Alpha:
	URL: /Users/chrisweber/Sites/ddev-mutagen
	Configuration:
		Watch mode: Default (Portable)
		Watch polling interval: Default (10 seconds)
		Probe mode: Default (Probe)
		Scan mode: Default (Accelerated)
		Stage mode: Neighboring
		File mode: Default (0600)
		Directory mode: Default (0700)
		Default file/directory owner: Default
		Default file/directory group: Default
	Connected: Yes
Beta:
	URL: docker://ddev-ddev-mutagen-web/var/www/html
		DOCKER_HOST=unix:///Users/chrisweber/.colima/default/docker.sock
	Configuration:
		Watch mode: Default (Portable)
		Watch polling interval: Default (10 seconds)
		Probe mode: Default (Probe)
		Scan mode: Default (Accelerated)
		Stage mode: Neighboring
		File mode: Default (0600)
		Directory mode: Default (0700)
		Default file/directory owner: Default
		Default file/directory group: Default
	Connected: No
Last error: beta polling error: unable to receive poll response: unable to read message length: unexpected EOF
Status: Connecting to beta
--------------------------------------------------------------------------------
Name: ddev-nomutagen
Identifier: sync_kdkVUfgta0aBSr4DOAaRkqYeDFmm0maB2IJ4RVwE47W
Labels:
	com.ddev.volume-signature: var-run-1677791725
Configuration:
	Synchronization mode: Two Way Resolved
	Maximum allowed entry count: Default (2⁶⁴−1)
	Maximum staging file size: Default (18 EB)
	Symbolic link mode: POSIX Raw
	Ignore VCS mode: Default (Propagate)
	Ignores:
		/.git
		/.tarballs
		/.ddev/db_snapshots
		/.ddev/.importdb*
		.DS_Store
		.idea
		/web/sites/default/files
	Permissions mode: Default (Portable)
Alpha:
	URL: /Users/chrisweber/Sites/ddev-nomutagen
	Configuration:
		Watch mode: Default (Portable)
		Watch polling interval: Default (10 seconds)
		Probe mode: Default (Probe)
		Scan mode: Default (Accelerated)
		Stage mode: Neighboring
		File mode: Default (0600)
		Directory mode: Default (0700)
		Default file/directory owner: Default
		Default file/directory group: Default
	Connected: Yes
	Synchronizable contents:
		6499 directories
		25372 files (126 MB)
		4 symbolic links
Beta:
	URL: docker://ddev-ddev-nomutagen-web/var/www/html
		DOCKER_HOST=unix:///var/run/docker.sock
	Configuration:
		Watch mode: Default (Portable)
		Watch polling interval: Default (10 seconds)
		Probe mode: Default (Probe)
		Scan mode: Default (Accelerated)
		Stage mode: Neighboring
		File mode: Default (0600)
		Directory mode: Default (0700)
		Default file/directory owner: Default
		Default file/directory group: Default
	Connected: Yes
	Synchronizable contents:
		6499 directories
		25372 files (126 MB)
		4 symbolic links
Status: Watching for changes
--------------------------------------------------------------------------------
======= Docker Info =========
Docker platform: docker-desktop 
Using docker context: default (unix:///var/run/docker.sock) 
docker-compose: v2.14.0 
Using DOCKER_HOST=unix:///var/run/docker.sock 
Docker version: 20.10.23 
Able to run simple container that mounts a volume. 
Able to use internet inside container. 
Docker disk space:
Filesystem                Size      Used Available Use% Mounted on
overlay                  58.4G     13.7G     41.6G  25% /

Container ddev-ddev-nomutagen-db  Removed 
Container ddev-ddev-nomutagen-dba  Removed 
Container ddev-ddev-nomutagen-web  Removed 
Container ddev-ddev-nomutagen-selenium-chrome  Removed 
Network ddev-ddev-nomutagen_default  Removed 
Project ddev-nomutagen has been stopped. 
Stopped mutagen daemon 
The ddev-ssh-agent container has been removed. When you start it again you will have to use 'ddev auth ssh' to provide key authentication again. 
Existing docker containers: 
CONTAINER ID   IMAGE                      COMMAND                  CREATED        STATUS                      PORTS     NAMES
46736414225d   docksal/varnish:6.6-3.1    "docker-entrypoint.s…"   29 hours ago   Exited (0) 26 hours ago               demo_varnish_1
d86e6faa7bdc   docksal/nginx:1.20-1.2     "docker-entrypoint.s…"   29 hours ago   Exited (0) 26 hours ago               demo_web_1
053f23225fe7   demo_cli                   "/opt/startup.sh sup…"   29 hours ago   Exited (0) 26 hours ago               demo_cli_1
40ba7f3c3186   b9af201457e0               "/bin/sh"                29 hours ago   Created                               phpstorm_helpers_PS-223.8617.59
3bbc4961fb6a   docksal/solr:1.0-solr3     "java -jar start.jar"    29 hours ago   Exited (137) 26 hours ago             demo_solr_1
a416b548745e   docksal/mariadb:10.4-1.3   "docker-entrypoint.s…"   29 hours ago   Exited (0) 26 hours ago               demo_db_1
0141f92dad8b   wodby/redis:5.0            "/docker-entrypoint.…"   29 hours ago   Exited (0) 26 hours ago               demo_redis_1
6bc888e99297   docksal/ssh-agent:1.4      "docker-entrypoint.s…"   7 days ago     Exited (2) 26 hours ago               docksal-ssh-agent
5ca1cc835e7c   docksal/dns:1.2            "docker-entrypoint.s…"   7 days ago     Exited (0) 26 hours ago               docksal-dns
7157df8b8a96   docksal/vhost-proxy:1.8    "docker-entrypoint.s…"   7 days ago     Exited (0) 26 hours ago               docksal-vhost-proxy
Creating a new ddev project config in the current directory (/Users/chrisweber/Sites/tryddevproject-16641) 
Once completed, your configuration will be written to /Users/chrisweber/Sites/tryddevproject-16641/.ddev/config.yaml
 
Configuring unrecognized codebase as project type 'php' at /Users/chrisweber/Sites/tryddevproject-16641/web 
Configuration complete. You may now run 'ddev start'. 
Network ddev_default created 
Starting tryddevproject-16641... 
Container ddev-ssh-agent  Started 
ssh-agent container is running: If you want to add authentication to the ssh-agent container, run 'ddev auth ssh' to enable your keys. 
Network ddev-tryddevproject-16641_default  Created 
Container ddev-tryddevproject-16641-web  Started 
Container ddev-tryddevproject-16641-db  Started 
Container ddev-tryddevproject-16641-dba  Started 
Container ddev-router  Started 
Successfully started tryddevproject-16641 
Project can be reached at https://tryddevproject-16641.ddev.site https://127.0.0.1:51837 
======== Curl of site from inside container:
HTTP/1.1 200 OK
Server: nginx
Date: Thu, 02 Mar 2023 21:25:54 GMT
Content-Type: text/html; charset=UTF-8
Connection: keep-alive
Vary: Accept-Encoding

======== curl -I of http://tryddevproject-16641.ddev.site from outside:
HTTP/1.1 200 OK
Server: nginx/1.20.1
Date: Thu, 02 Mar 2023 21:25:54 GMT
Content-Type: text/html; charset=UTF-8
Connection: keep-alive
Vary: Accept-Encoding

======== full curl of http://tryddevproject-16641.ddev.site from outside:
Success accessing database... db via TCP/IP
ddev is working. You will want to delete this project with 'ddev delete -Oy tryddevproject-16641'
======== Project ownership on host:
drwxr-xr-x  4 chrisweber  staff  128 Mar  2 15:25 ../tryddevproject-16641
======== Project ownership in container:
drwxr-xr-x 4 chrisweber dialout 128 Mar  2 21:25 /var/www/html
======== In-container filesystem:
Filesystem     Type          1K-blocks      Used Available Use% Mounted on
grpcfuse       fuse.grpcfuse 971350180 237141232 734208948  25% /var/www/html
======== curl again of tryddevproject-16641 from host:
Success accessing database... db via TCP/IP
ddev is working. You will want to delete this project with 'ddev delete -Oy tryddevproject-16641'
Thanks for running the diagnostic. It was successful.
Please provide the output of this script in a new gist at gist.github.com
Running ddev launch in 5 seconds
If you're brave and you have jq you can delete all tryddevproject instances with this one-liner:
    ddev delete -Oy $(ddev list -j |jq -r .raw[].name | grep tryddevproject)
In the future ddev debug test will also provide this option.

Please delete this project after debugging with 'ddev delete -Oy tryddevproject-16641'
```

</details>

### Expected Behavior

As a developer, I want to run + debug tests using PhpStorm's test runner.

### Actual Behavior

I am able to run tests via the command line, but I can't run tests using PhpStorm's test runner when mutagen is enabled.

If I disable mutagen, then I can run and debug tests using PhpStorm's test runner.

### Steps To Reproduce

# DDEV + Mutagen caused PhpStorm errors when running tests
### 1. Use CMS Quickstart to build + install PHPUnit

Using Docker Desktop:
```bash
mkdir my-drupal10-site
cd my-drupal10-site
ddev config --project-type=drupal10 --docroot=web --create-docroot

# Enable Mutagen:  This part introduces the bug.
ddev config --mutagen-enabled=true

ddev start
ddev composer create drupal/recommended-project
ddev composer require drupal/core-dev --dev --update-with-all-dependencies
ddev composer require drush/drush
ddev drush site:install --account-name=admin --account-pass=admin -y
```
### 2. Run a basic Unit Test | this should work.
```bash
ddev exec phpunit -c /var/www/html/web/core/phpunit.xml.dist /var/www/html/web/core/modules/system/tests/src/Unit/Routing/AdminRouteSubscriberTest.php
```
OR like this:
```bash
ddev exec -d /var/www/html/web "../vendor/bin/phpunit -v -c ./core/phpunit.xml.dist ./core/modules/system/tests/src/Unit/Routing/AdminRouteSubscriberTest.php"
```

### 3. Run the same tests in PhpStorm
Navigate to core/modules/system/tests/src/Unit/Routing/AdminRouteSubscriberTest.php and run the test by click on the test icon next to either a method or class.

The error you will receive will be: note I replaced my user directory with `<user-dir>`
```
Test framework quit unexpectedly

[docker-compose://[/Users/<user-dir>/Sites/ddev-mutagen/.ddev/.ddev-docker-compose-full.yaml]:web/]:php8.1 vendor/bin/phpunit --configuration /var/www/html/web/core/phpunit.xml.dist --filter Drupal\\Tests\\system\\Unit\\Routing\\AdminRouteSubscriberTest --test-suffix AdminRouteSubscriberTest.php /Users/<user-dir>/Sites/ddev-mutagen/web/core/modules/system/tests/src/Unit/Routing --teamcity
Testing started at 8:44 AM ...
PHPUnit 9.6.4 by Sebastian Bergmann and contributors.

Cannot open file "/Users/<user-dir>/Sites/ddev-mutagen/web/core/modules/system/tests/src/Unit/Routing".

Process finished with exit code 1
```

### 4. Disable Mutagen
```bash
ddev config --mutagen-enabled=false
ddev start
```

### 5. Run the same tests in PhpStorm, again
You can run the test on the command line, it will continue to work.
```bash
ddev exec phpunit -c /var/www/html/web/core/phpunit.xml.dist /var/www/html/web/core/modules/system/tests/src/Unit/Routing/AdminRouteSubscriberTest.php
```
OR like this:
```bash
ddev exec -d /var/www/html/web "../vendor/bin/phpunit -v -c ./core/phpunit.xml.dist ./core/modules/system/tests/src/Unit/Routing/AdminRouteSubscriberTest.php"
```

Now, navigate to core/modules/system/tests/src/Unit/Routing/AdminRouteSubscriberTest.php and run the test by click on the test icon next to either a method that isn't setup or the class.

Running tests in PhpStorm should work now.**

### Anything else?

What we're likely seeing here is an error not with DDEV but with PhpStorm.  Still, I thought it prudent to create a bug report here to refer to the bug report I'll make with them.

I've created a full repo of my code that demonstrates this issue here: https://github.com/cosmicdreams/ddev-mutagen-phpstorm-error

Related PhpStorm error report: https://youtrack.jetbrains.com/issue/WI-71654/DDEV-Using-Mutagen-breaks-the-ability-to-run-tests-in-PhpStorm-not-the-command-line

**Comment by rfay** [comment]
I see what's going on. PhpStorm is suggesting the wrong path mappings when mutagen is enabled. 

Change the path mappings from /var/tmp/html to /var/www/html.

`<Project root>→/var/tmp/html; /Users/rfay/Library/Application Support/JetBrains/PhpSto...`

**Comment by cosmicdreams** [comment]
That did indeed fix my issue.  This is such a huge win for those of us that use PhpStorm.  These tests are running VERY fast now.

**Comment by cosmicdreams** [comment]
Now that we see this what are the Next Steps? 

DDev documentation fix: https://github.com/ddev/ddev/pull/4693
[PhpStorm bug tracking](https://youtrack.jetbrains.com/issue/WI-71654/DDEV-Using-Mutagen-breaks-the-ability-to-run-tests-in-PhpStorm-not-the-command-line)
File an issue with the DDEV PhpStorm plugin: https://github.com/php-perfect/ddev-intellij-plugin/issues/166

**Comment by rfay** [comment]
Replied on the PhpStorm issue; he'll need more help to understand. Hopefully my reply will help.

The key in the plugin issue is whether that's even in scope for the plugin, whether it manages the default mappings at all.

**Comment by rfay** [comment]
It seems like everything is in its proper place now. Can't remember if the mapping problem is already in docs or if it's in your PR now.

**Comment by cosmicdreams** [comment]
I hadn't noticed their reply.  I added testing steps in the form of sample tests to run.

**Comment by rfay** [comment]
Make sure you "watch" the bug at https://youtrack.jetbrains.com/issue/WI-71654/DDEV-Using-Mutagen-breaks-the-ability-to-run-tests-in-PhpStorm-not-the-command-line - requires a login via github or otherwise.

**Comment by rfay** [comment]
Actually... I didn't get an email notification of your post, even though I'm watching. They may have something broken.

**Comment by rfay** [comment]
One approach here that may help with this is to remove the internally-used /var/tmp/html mount (for other reasons) and maybe then PhpStorm won't be confused. I may well work on that. The internal reasons for its existence are complex, and finding a more sustainable technique is appealing.

**Comment by rfay** [comment]
This has been solved in DDEV v1.22.0-beta*, and has been dealt with in the PHPStorm extension as well. Would very much appreciate your confirmation @cosmicdreams ! Closing for now.
